tests/language/expressions/operand-gc-roots/operand-gc-roots.js

```
@@ -0,0 +1,244 @@
+
+  test("both operands coercing keeps the left result alive", () => {
+    // The left operand's coerced value is itself a fresh allocation, and the
+    // right operand's hook collects after it exists: 9 * 4 answered 16 when it
+    // was unrooted, because the left had been replaced by the right.
+    expect(box(9) * box(4)).toBe(36);
+    expect(box(9) - box(4)).toBe(5);
+    expect(box(9) + box(4)).toBe(13);
+    expect(box(9) / box(4)).toBe(2.25);
+    expect(box(9) < box(4)).toBe(false);
+    expect(box(4) < box(9)).toBe(true);
+    expect(box(9) == box(9)).toBe(false); // distinct objects, ES2026 §7.2.13
```

Biome (2.5.6)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)